research/2020-36-json-ld-and-rdfa.md

```diff
@@ -0,0 +1,486 @@
+* Create a JavaScript function that will be able to update the existing in page JSON-LD anchor local reference by the corresponding RDF content marked up in RDFa.
+* Use the same JSON-LD RDF main anchor as the one for the main content of the page. Both will use "#wb-main" as both define content published in it's web page document container.
+
+Example (Page URL: http://example.com)
+```
+<script id="wb-script" type="application/ld+json">
+
+    {
+	"@context": {
+		"@vocab": "http://schema.org/"
+	},
+	"@id": "#wb-main",
+	"@type":"Thing",
+	"publisher": {
+    	"@id": "#wb-publisher"
+    }
+}
+</script>
+
+<body vocab="http://schema.org/" typeof="WebPage">
+	<div typeof="Organisation" resource="#wb-publisher">
+		<img src="/link/to/a/logo" alt="Publisher name" property="logo" />
+		<meta property="name" content="Publisher name" />
+	</div>
+
+	<main property="mainContentOfPage" typeof="WebPageElement" resource="#wb-main">
+
+		<h1 property="name">Example</h1>
+		<p property="description">Description for the main content of the page</p>
+
+		... Other content tagged with RDFa ...
+
+	</main>
+</body>
+```
+
+With the objective to produce the following triple, expressed in Turtle
+```
+<http://example.com#wb-publisher>
+	a <http://schema.org/Organisation> ;
+	<http://schema.org/logo> <http://example.com/link/to/a/logo> ;
+	<http://schema.org/name> "Publisher name" .
+
+<http://example.com#wb-main>
+	a <http://schema.org/Thing>, <http://schema.org/WebPageElement> ;
+	<http://schema.org/publisher> <http://example.com#wb-publisher> ;
+	<http://schema.org/name> "Example" ;
+	<http://schema.org/description> "Description for the main content of the page" .
+```
+
+The JSON-LD content can be specifically targeted from this URL: <http://example.com#wb-script>
+
+
+### Findings assumed for Google
+
+The following markup and strategy are:
+
+* Acceptable: Only RDFa
+* Acceptable: Only JSON-LD
+* Acceptable: Multiple JSON-LD script element. They get merged into one and same dataset. As defined by JSON-LD section 7 and illustrated by the example #145
+* Seems not supported: RDFa + JSON-LD where their content crisscross. For example, when I did my test/research, the RDF anchor was not taken in consideration.
+* Worth to give a try: Support the RDFa + JSON-LD for content complementary of the main content in order to avoid content duplication. For example the publisher information is available in RDFa markup in the template and the main thing described in JSON-LD would refer to the main page publisher instead of duplicating the information.
+
+The "Seems not supported" option can probably by fixed by a javascript procedure that will complete the JSON-LD content and do automated duplication. That way, it will reduce the risk of human error like copy and paste or forgetting to update the information in several places. That became riskier when there is a large publishing team and the content need to go through a complex publishing and approval workflow.
```

Comment:
```suggestion
The "Seems not supported" option can probably by fixed by a javascript procedure that will complete the JSON-LD content and do automated duplication. That way, it will reduce the risk of human error like copy and paste or forgetting to update the information in several places. That becomes riskier when there is a large publishing team and the content need to go through a complex publishing and approval workflow.
```